lib/api/v3/users/users_api.rb

```diff
@@ -52,6 +52,42 @@ class UsersAPI < Grape::API
                 fail ::API::Errors::Unauthorized
               end
             end
+
+            namespace :lock do
+
+              # Authenticate lock transitions
+              before do
+                if !User.current.admin?
+                  fail ::API::Errors::Unauthorized
+                end
+              end
+
+              desc 'Set lock on user account'
+              post do
+                # Silently ignore lock -> lock transition
+                if @user.active? || @user.locked?
+                  @user.lock! unless @user.locked?
+
+                  status 200
+                  UserRepresenter.new(@user)
+                else
+                  fail ::API::Errors::InvalidUserStatusTransition
+                end
+              end
+
+              desc 'Remove lock on user account'
+              delete do
+                # Silently ignore active -> active transition
+                if @user.locked? || @user.active?
+                  @user.activate! unless @user.active?
+
+                  status 200
+                  UserRepresenter.new(@user)
+                else
+                  fail ::API::Errors::InvalidUserStatusTransition
+                end
+              end
```

Comment:
The ```current_user``` is not provided for the representer. Therefore, the links for locking/unlocking will always be missing in the response returned from those two endpoints. 

Changing this to

```UserRepresenter.new(@user, current_user: curren_user)```

Will help here. You do it correctly in the representer specs.

The GET action for user is also missing the current_user. Due to that, the link for locking/unlocking is currently displayed nowhere.

Additionally, but arguably outside of the scope of this PR, it should be added for ```available_(assignees|responsibles)``` as well.